common/src/sql/dbinit.sql

```diff
@@ -984,6 +987,15 @@ CREATE TABLE omicron.public.ip_pool_range (
     /* The range is inclusive of the last address. */
     last_address INET NOT NULL,
     ip_pool_id UUID NOT NULL,
+    /* Optional ID of the project for which this range is reserved.
+     *
+     * NOTE: This denormalizes the tables a bit, since the project_id is
+     * duplicated here and in the parent `ip_pool` table. We're allowing this
+     * for now, since it reduces the complexity of the already-bad IP allocation
+     * query, but we may want to revisit that, and JOIN with the parent table
+     * instead.
+     */
+    project_id UUID,
```

Comment:
Do we have an issue to capture this rework?

Reply:
Which rework are you referring to?

Reply:
It sounds like that's the real chunk of work to be done, so yes, that. I just worry something like this would be easy to lose track of.

Reply:
Will do!

Reply:
https://github.com/oxidecomputer/omicron/issues/1468